client/my-sites/theme/main.jsx

```diff
@@ -127,12 +129,13 @@ const ThemeSheet = React.createClass( {
 		};
 
 		const { siteSlug, id } = this.props;
+		const sitePart = siteSlug ? `/${ siteSlug }` : '';
 
 		const nav = (
 			<NavTabs label="Details" >
 				{ this.getValidSections().map( ( section ) => (
 					<NavItem key={ section }
-						path={ `/theme/${ id }/${ section }/${ siteSlug }` }
+						path={ `/theme/${ id }/${ section }${ sitePart }` }
```

Comment:
the / on the end of the path here (in cases with no site) was causing an unnecessary redirect.